### PR TITLE
Secure and consolidate announcements endpoints

### PR DESCRIPTION
**Pull request by exbinary**
### Why
Securing AnnouncementsController is necessary before we can re-enable user registration #840.

### What and How
- [x] Add `AnnouncementPolicy`
- [x] Wire policy into `AnnouncementsController`
- [x] Consolidate `/announcements` with `/announcements_list`. See screenshots below.
- [x] Change from `Time.now` to `Time.current` in `Announcement.published` and refactor the implementation of that scope while i was in there.

### Testing
- [x] announcement_policy_spec.rb
- [x] requests/announcements_spec.rb

### Next Steps
?

### Outstanding Questions, Concerns and Other Notes
* I'm not sure 'cards' are the best UI element in the consolidated /announcements index. Works for now but feels a bit heavy/unnecessary particularly with a long list?
* @maebeale , would love your thoughts on the consolidated /announcements view. The 'admin' info/links at the bottom right of each card are hidden for non-admins.
* ?

### Pre-Merge Checklist
- [x] All new features have been described in the pull request
- [x] Security & accessibility have been considered
- [x] High quality tests have been added, or an explanation has been given why the features cannot be tested
- [x] New features have been documented, and the code is understandable and well commented
- [ ] Entry added to CHANGELOG.md if appropriate
- [x] All outstanding questions and concerns have been resolved
- [x] Any next steps that seem like good ideas have been created as issues for future discussion & implementation

#### Unauthenticated (on iPad Pro)
Screen Shot 2021-01-28 at 10 37 18 AM
#### Admin (on iPhone 6)
Screen Shot 2021-01-28 at 10 36 08 AM

### CODE REVIEW

File: app/models/announcement.rb [hunk 1]
```diff
@@ -7,18 +7,24 @@ class Announcement < ApplicationRecord
...
+    approved
+      .where(publish_from: before_now, publish_until: nil).or(
+    approved
+      .where(publish_from: before_now, publish_until: after_now)
+    )
   end
 
   def published?
-    now = Time.now
+    now = Time.current
```
Comment (h-m-m) [inline app/models/announcement.rb:27]: 👍

File: app/views/announcements/index.html.erb [hunk 2]
```diff
@@ -1,30 +1,26 @@
...
-            <strong><%= announcement.name %></strong>
-          </td>
-          <td>
-            <strong><%= announcement.description&.truncate(20) %></strong>
-          </td>
-          <td class="has-text-centered"><%= yes_no(announcement.published?) %></td>
-          <td><%= edit_button(announcement) %></td>
-        </tr>
+<div class="row">
+  <% @announcements.each do |announcement| %>
```
Comment (h-m-m) [inline app/views/announcements/index.html.erb:4]: If this is not really a `<table>`, is it actually an `<ul>` instead?

**Review by rodri0315 — APPROVED** [review]
Thank you @h-m-m for reviewing this with me. Learned a lot!

File: app/views/announcements/index.html.erb (hunk 2 above)
Comment (exbinary) [reply app/views/announcements/index.html.erb]: Love that idea!

**Pull request merged**